lua/wire/client/hlzasm/hc_preprocess.lua

```diff
@@ -70,6 +70,56 @@ function HCOMP:ParsePreprocessMacro(lineText,macroPosition)
       CPULib.CPUName = pragmaCommand
     elseif pragmaName == "searchpath" then
       table.insert(self.SearchPaths,pragmaCommand)
+    elseif pragmaName == "allow" or pragmaName == "zap" then
+    if not self.Settings.AutoBusyRegisters then
+      self.Settings.AutoBusyRegisters = true
+    end
+    local StartRegister, EndRegister = string.match(macroParameters, "([^,%s]+)%s*,%s*([^,%s]+)")
+    if StartRegister == nil then
+      self:Error("Missing register range argument",
```

Comment:
Does self:Error throw an error? We might need to early return if not, otherwise this function will keep executing an run into a lua error.

Reply:
I believe error halts compilation entirely, even in cases where there should be multiple errors I only get the first error that it runs into.

Reply:
Looks like it does throw an error so it should be fine. Just make sure you test that it catches those bad cases.

Reply:
I tried doing cases where it checks endregister == nil but they all ran into startregister being nil instead, so I just left it as startregister and made the error for startregister being nil generic(as if the argument were missing entirely)